src/psij/executors/batch/batch_scheduler_executor.py

```diff
@@ -181,6 +181,7 @@ def submit(self, job: Job) -> None:
         self._ensure_work_dir()
         assert (job.spec)
 
+        job.executor = self
```

Comment:
Should we document that an executor implementation SHOULD NOT keep job references around - because otherwise we create circular references and make life difficult for the garbage collector.  

Or we could do something like this?
```py
job.set_executor(self)
class Job():
  ...
  def set_executor(self, executor):
      self._executor_ref = weakref.ref(executor)
  @property
  def executor(self):
      if self._executor_ref:
          return self._executor_ref()
```

Or maybe I am off target and circ refs cannot happen?

Reply:
I think the idea is that executors are more or less permanent objects: you instantiate them at some point and then you keep using them to submit jobs until your "workflow" ends. So my guess is that executors holding references to jobs is a problem with or without circular references.

Reply:
That's good enough for me.